cpp/pybind/t/pipelines/registration/registration.cpp

```diff
@@ -0,0 +1,256 @@
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#include "open3d/t/pipelines/registration/Registration.h"
+
+#include <memory>
+#include <utility>
+
+#include "open3d/t/geometry/PointCloud.h"
+#include "open3d/t/pipelines/registration/TransformationEstimation.h"
+#include "open3d/utility/Console.h"
+#include "pybind/docstring.h"
+#include "pybind/t/pipelines/registration/registration.h"
+
+namespace open3d {
+namespace t {
+namespace pipelines {
+namespace registration {
+
+template <class TransformationEstimationBase = TransformationEstimation>
+class PyTransformationEstimation : public TransformationEstimationBase {
+public:
+    using TransformationEstimationBase::TransformationEstimationBase;
+    TransformationEstimationType GetTransformationEstimationType() const {
+        PYBIND11_OVERLOAD_PURE(TransformationEstimationType,
+                               TransformationEstimationBase, void);
+    }
+    double ComputeRMSE(const t::geometry::PointCloud &source,
+                       const t::geometry::PointCloud &target,
+                       const CorrespondenceSet &corres) const {
+        PYBIND11_OVERLOAD_PURE(double, TransformationEstimationBase, source,
+                               target, corres);
+    }
+    core::Tensor ComputeTransformation(const t::geometry::PointCloud &source,
+                                       const t::geometry::PointCloud &target,
+                                       const CorrespondenceSet &corres) const {
+        PYBIND11_OVERLOAD_PURE(core::Tensor, TransformationEstimationBase,
+                               source, target, corres);
+    }
+};
+
+void pybind_registration_classes(py::module &m) {
+    // open3d.t.pipelines.registration.ICPConvergenceCriteria
+    py::class_<ICPConvergenceCriteria> convergence_criteria(
+            m, "ICPConvergenceCriteria",
+            "Class that defines the convergence criteria of ICP. "
+            "ICP algorithm stops if the relative change of fitness and rmse "
+            "hit ``relative_fitness`` and ``relative_rmse`` individually, "
+            "or the iteration number exceeds ``max_iteration``.");
+    py::detail::bind_copy_functions<ICPConvergenceCriteria>(
+            convergence_criteria);
+    convergence_criteria
+            .def(py::init([](double fitness, double rmse, int itr) {
```

Comment:
If the constructor signature is exactly the same across python and c++ (e.g. we don't need any conversions), it should be possible to use `py::init<xxx, xxx>()` with the types directly. In this case, it will look like
```cpp
.def(py::init<double, double, int>(), "relative_fitness"_a = 1e-6,
     "relative_rmse"_a = 1e-6, "max_iteration"_a = 30)
```
Could you check if it works properly and apply to the rest of the constructors in this PR?

Reply:
Interesting. This works, thanks. I have also used the same for the default non-parameterized constructor. 
For future reference, can we also use `vector<double>`, `core::Tensor`, `t::geometry::PointCloud`, `core::Device`, `core::Dtype` in place of `xxx` in `py::init<xxx, xxx>()` ?

Reply:
Done.